src/problems/Albany_ThermalProblem.hpp

```diff
@@ -115,8 +114,9 @@ namespace Albany {
 #include "Albany_ResponseUtilities.hpp"
 //#include "PHAL_Neumann.hpp"
 #include "PHAL_ThermalResid.hpp"
-#include "PHAL_ThermalSource.hpp"
-
+//IKT FIXME - add dependence on LandIce
+#include "../../LandIce/evaluators/LandIce_SharedParameter.hpp"
+#include "../../LandIce/problems/LandIce_ParamEnum.hpp"
 
```

Comment:
This could create problems with the libraries that have been just cleaned up by Luca. I would give it a try to the PHAL_ShareParameters, so that you don't have to depend on LandIce, it is much cleaner.

Reply:
So, I started with using PHAL_ShareParameters but I couldn't get it to work correctly.  I spent a few hours trying to figure it out.  Then when I switched to the land ice shared parameter, it worked right away.  I figured it wasn't worth my time to figure out what I was doing wrong with the shared parameters since that wasn't really relevant to the sensitivity calculation itself.  The fact that we care about transient sensitivities for LandIce, where we'd be using the LandIce::SharedParameter, solidified this decision.  I did put ifdef guards to prevent configuration errors in the case LandIce is not enabled.  I could move the problem inside the LandIce directory if you think that'll be less confusing.  If you really think it's worth taking the time to figure out why the PHAL_ShareParameters way was not working, I could try to do that.  I thought it wasn't worth the effort.

Reply:
I see. We could always create a Thermal::SharedParameter class, basically copying the LandIce one. In this way we would avoid inter-dependencies between albanyLib and landIceLib. Would that make sense to you? Any comment on this @bartgol?

Reply:
I like this idea, but lets see what Luca says.

Reply:
We can also copy this version of SharedParameter back into PHAL. Not sure if we should replace the old one or add it along side it. If Irina had some issues with the old one, then it might be worth checking if something is really intrinsically wrong there, and, if so, maybe toss it and keep this.

I'm not a huge fan of the LandIce::SharedParameter class. I wrote it cause I was having some inconsistency of parameters across different field managers, but it might be I was using the PHAL one wrong. The extra template parameters in the class make it a bit stiff, and hard to use, so making the PHAL one work would be much better.

Either way, I don't think including LandIce stuff in the main albany lib is a good pattern, even if ifdeffed out. We spent some effort cleaning up cyclic deps, and this feels like a step back. As I said, migrating LandIce::SharedParameter to PHAL (perhaps as GlobalSharedParameter or something, given its static members) is definitely more appropriate.

Reply:
Notice that porting LandIce::SharedParameter into PHAL would be fine. The class's two last template parameters are an enum type and an enum value. So one can use it with LandIce::ParamEnum, as well as with, say, ThermalParamEnum.

Reply:
Makes sense - thanks Luca.  I'll add the new class and take out the ifdef guards around the Thermal Problem since it will no longer depend on ALI.  I like this solution.

Reply:
I just pushed some changes that rename LandIce::SharedParameter PHAL::SharedParameter.  This allows one to remove the LandIce dependency of ThermalProblem.  Much better!  I will mark this as resolved too.

Reply:
That's great. thanks. What about the old PHAL::SharedParameters in evaluators/utility? Did you remeve it?

Reply:
We should not remove it, imho. That file also contains SharedParameterVec, which allows to use one evaluator for a vector of parameters. The version in LandIce_SharedParameter.hpp does not support that (although, it shouldn't be hard to add it). I'm not sure if SharedParameterVec is used at all (although it seems like ThermalProblem could use it), but I would keep that file. I understand it would generate a name clashing between old and new SharedParameter, which is why I would call the new one something like GlobalSharedParametter or StaticSharedParameter (though static has N+1 meanings and might be confusing).

Reply:
I forgot in my initial commit, but I just did in a second commit.  It was not being used anyway, and when I tried to use it, I had issues, so it's probably best to remove it in favor of the new PHAL::SharedParameter.  Do you think the evaluators/utility dir is a better place for it than evaluators/pde?

Reply:
By "it" I mean the new PHAL_SharedParameter that replaces LandIce_SharedParameter.

Reply:
@bartgol : shoot I pushed my changes before seeing your comment.  The old PHAL::SharedParameter class was not used anywhere.  I'm not sure what to do...  I can bring it back and give the new SharedParameter a different name, maybe, as you suggest?

Reply:
If it wasn't used, then it's fine, I guess. I thought it was, since we do have optimization tests in main albany, but I see now that their parameters are in the DBCs, and Dirichlet evaluators also act as SharedParameter evaluators (not sure if this is the best approach, since it's mixing apples and oranges in the same class, but this is not the time and place to discuss that).

If you can, put a TODO comment in the (new) SharedParameter evaluator, reminding us to implement support for vector parameters.

Reply:
I will do this, Luca.  Do you think it's better for the new SharedParameter file to live in evaluators/pde or evaluators/utility?  I put it in the former, but maybe the latter is better?

Reply:
> Do you think the evaluators/utility dir is a better place for it than evaluators/pde?

Yes, I think it should be located in `evaluators/utility`.

Reply:
I have moved it and added the comment about the TODO.  I will mark this thread re-resolved.